### ISSUE TITLE
 bug | urllib upgrade breaks new OpenKAT install | (Rocky,Octopoes,Boefjes,)

### ISSUE THREAD

**Issue by tobiasBDO** [minvws/nl-kat-coordination#984]
_Please add `bug`, the name of any relevant modules (e.g. `rocky`), and any other relevant labels to your issue._

**Describe the bug**
Seems like this commit https://github.com/minvws/nl-kat-coordination/commit/3fdfe4df58c6420e533e0945a31dc712b1aa8848 which upgrades requests from 2.29.0 to 2.30.0 breaks communication in between OpenKAT containers.

Relevant discussion: https://github.com/docker/docker-py/issues/3113

**To Reproduce**
1. Clone main
2. `make kat`
3. onboard organization
4. start boefje

**Expected behavior**
I expect the boefje to grab the requested data and pass it to a normalizer.

**OpenKAT version**
OpenKAT main on `Ubuntu 20.04` with `Docker version 23.0.6, build ef23cbc`

**Current behavior**

***rocky***
```
rocky_1                | DEBUG:urllib3.connectionpool:http://scheduler:8000 "GET /tasks?limit=0&scheduler_id=boefje-demo&type=boefje HTTP/1.1" 200 246
rocky_1                | DEBUG:urllib3.connectionpool:http://scheduler:8000 "GET /tasks?limit=2&offset=0&scheduler_id=boefje-demo&type=boefje HTTP/1.1" 200 1492
rocky_1                | [16/May/2023 11:14:43] "GET /nl/demo/tasks/ HTTP/1.0" 200 9757
rocky_1                | Operations to perform:
rocky_1                |   Apply all migrations: account, admin, auth, contenttypes, fmea, otp_static, otp_totp, password_history, sessions, tools, two_factor
rocky_1                | Running migrations:
rocky_1                |   No migrations to apply.
rocky_1                |   Your models in app(s): 'two_factor' have changes that are not yet reflected in a migration, and so won't be applied.
rocky_1                |   Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage.py migrate' to apply them.
rocky_1                | Watching for file changes with StatReloader
rocky_1                | INFO:django.utils.autoreload:Watching for file changes with StatReloader
rocky_1                | Performing system checks...
rocky_1                |
rocky_1                | System check identified no issues (0 silenced).
rocky_1                | May 16, 2023 - 11:16:23
rocky_1                | Django version 3.2.19, using settings 'rocky.settings'
rocky_1                | Starting development server at http://0.0.0.0:80/
rocky_1                | Quit the server with CONTROL-C.
rocky_1                | DEBUG:urllib3.connectionpool:Starting new HTTP connection (1): scheduler:8000
rocky_1                | Internal Server Error: /nl/demo/tasks/
rocky_1                | Traceback (most recent call last):
rocky_1                |   File "/usr/local/lib/python3.8/site-packages/urllib3/connection.py", line 200, in _new_conn
rocky_1                |     sock = connection.create_connection(
rocky_1                |   File "/usr/local/lib/python3.8/site-packages/urllib3/util/connection.py", line 85, in create_connection
rocky_1                |     raise err
rocky_1                |   File "/usr/local/lib/python3.8/site-packages/urllib3/util/connection.py", line 73, in create_connection
rocky_1                |     sock.connect(sa)
rocky_1                | ConnectionRefusedError: [Errno 111] Connection refused
rocky_1                |
rocky_1                | The above exception was the direct cause of the following exception:
rocky_1                |
rocky_1                | Traceback (most recent call last):
rocky_1                |   File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 790, in urlopen
rocky_1                |     response = self._make_request(
rocky_1                |   File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 496, in _make_request
rocky_1                |     conn.request(
rocky_1                |   File "/usr/local/lib/python3.8/site-packages/urllib3/connection.py", line 388, in request
rocky_1                |     self.endheaders()
rocky_1                |   File "/usr/local/lib/python3.8/http/client.py", line 1251, in endheaders
rocky_1                |     self._send_output(message_body, encode_chunked=encode_chunked)
rocky_1                |   File "/usr/local/lib/python3.8/http/client.py", line 1011, in _send_output
rocky_1                |     self.send(msg)
rocky_1                |   File "/usr/local/lib/python3.8/http/client.py", line 951, in send
rocky_1                |     self.connect()
rocky_1                |   File "/usr/local/lib/python3.8/site-packages/urllib3/connection.py", line 236, in connect
rocky_1                |     self.sock = self._new_conn()
rocky_1                |   File "/usr/local/lib/python3.8/site-packages/urllib3/connection.py", line 215, in _new_conn
rocky_1                |     raise NewConnectionError(
rocky_1                | urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f77cc43cdf0>: Failed to establish a new connection: [Errno 111] Connection refused
rocky_1                |
rocky_1                | The above exception was the direct cause of the following exception:
rocky_1                |
rocky_1                | Traceback (most recent call last):
rocky_1                |   File "/usr/local/lib/python3.8/site-packages/requests/adapters.py", line 486, in send
rocky_1                |     resp = conn.urlopen(
rocky_1                |   File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 844, in urlopen
rocky_1                |     retries = retries.increment(
rocky_1                |   File "/usr/local/lib/python3.8/site-packages/urllib3/util/retry.py", line 515, in increment
rocky_1                |     raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]
rocky_1                | urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='scheduler', port=8000): Max retries exceeded with url: /tasks?limit=0&scheduler_id=boefje-demo&type=boefje (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f77cc43cdf0>: Failed to establish a new connection: [Errno 111] Connection refused'))
rocky_1                |
rocky_1                | During handling of the above exception, another exception occurred:
rocky_1                |
rocky_1                | Traceback (most recent call last):
rocky_1                |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
rocky_1                |     response = get_response(request)
rocky_1                |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
rocky_1                |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
rocky_1                |   File "/usr/local/lib/python3.8/site-packages/django/views/generic/base.py", line 70, in view
rocky_1                |     return self.dispatch(request, *args, **kwargs)
rocky_1                |   File "/usr/local/lib/python3.8/site-packages/django/utils/decorators.py", line 43, in _wrapper
rocky_1                |     return bound_method(*args, **kwargs)
rocky_1                |   File "/usr/local/lib/python3.8/site-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view
rocky_1                |     return view_func(request, *args, **kwargs)
rocky_1                |   File "/usr/local/lib/python3.8/site-packages/django/utils/decorators.py", line 43, in _wrapper
rocky_1                |     return bound_method(*args, **kwargs)
rocky_1                |   File "/usr/local/lib/python3.8/site-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view
rocky_1                |     return view_func(request, *args, **kwargs)
rocky_1                |   File "/usr/local/lib/python3.8/site-packages/django/views/generic/base.py", line 98, in dispatch
rocky_1                |     return handler(request, *args, **kwargs)
rocky_1                |   File "/app/rocky/rocky/views/tasks.py", line 53, in get
rocky_1                |     return super().get(request, *args, **kwargs)
rocky_1                |   File "/usr/local/lib/python3.8/site-packages/django/views/generic/list.py", line 157, in get
rocky_1                |     context = self.get_context_data()
rocky_1                |   File "/app/rocky/rocky/views/tasks.py", line 79, in get_context_data
rocky_1                |     context = super().get_context_data(**kwargs)
rocky_1                |   File "/app/rocky/account/mixins.py", line 100, in get_context_data
rocky_1                |     context = super().get_context_data(**kwargs)
rocky_1                |   File "/usr/local/lib/python3.8/site-packages/django/views/generic/list.py", line 119, in get_context_data
rocky_1                |     paginator, page, queryset, is_paginated = self.paginate_queryset(queryset, page_size)
rocky_1                |   File "/usr/local/lib/python3.8/site-packages/django/views/generic/list.py", line 69, in paginate_queryset
```

***boefje***

```
boefje_1               | Traceback (most recent call last):
boefje_1               |   File "/usr/local/lib/python3.8/site-packages/docker/api/client.py", line 214, in _retrieve_server_version
boefje_1               |     return self.version(api_version=False)["ApiVersion"]
boefje_1               |   File "/usr/local/lib/python3.8/site-packages/docker/api/daemon.py", line 181, in version
boefje_1               |     return self._result(self._get(url), json=True)
boefje_1               |   File "/usr/local/lib/python3.8/site-packages/docker/utils/decorators.py", line 46, in inner
boefje_1               |     return f(self, *args, **kwargs)
boefje_1               |   File "/usr/local/lib/python3.8/site-packages/docker/api/client.py", line 237, in _get
boefje_1               |     return self.get(url, **self._set_request_timeout(kwargs))
boefje_1               |   File "/usr/local/lib/python3.8/site-packages/requests/sessions.py", line 600, in get
boefje_1               |     return self.request("GET", url, **kwargs)
boefje_1               |   File "/usr/local/lib/python3.8/site-packages/requests/sessions.py", line 587, in request
boefje_1               |     resp = self.send(prep, **send_kwargs)
boefje_1               |   File "/usr/local/lib/python3.8/site-packages/requests/sessions.py", line 701, in send
boefje_1               |     r = adapter.send(request, **kwargs)
boefje_1               |   File "/usr/local/lib/python3.8/site-packages/requests/adapters.py", line 486, in send
boefje_1               |     resp = conn.urlopen(
boefje_1               |   File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 790, in urlopen
boefje_1               |     response = self._make_request(
boefje_1               |   File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 496, in _make_request
boefje_1               |     conn.request(
boefje_1               | TypeError: request() got an unexpected keyword argument 'chunked'
boefje_1               |
boefje_1               | During handling of the above exception, another exception occurred:
boefje_1               |
boefje_1               | Traceback (most recent call last):
boefje_1               |   File "/app/boefjes/boefjes/local.py", line 50, in run
boefje_1               |     return boefje_resource.module.run(boefje_meta)
boefje_1               |   File "/app/boefjes/boefjes/plugins/kat_amass_brute/main.py", line 19, in run
boefje_1               |     client = docker.from_env()
boefje_1               |   File "/usr/local/lib/python3.8/site-packages/docker/client.py", line 96, in from_env
boefje_1               |     return cls(
boefje_1               |   File "/usr/local/lib/python3.8/site-packages/docker/client.py", line 45, in __init__
boefje_1               |     self.api = APIClient(*args, **kwargs)
boefje_1               |   File "/usr/local/lib/python3.8/site-packages/docker/api/client.py", line 197, in __init__
boefje_1               |     self._version = self._retrieve_server_version()
boefje_1               |   File "/usr/local/lib/python3.8/site-packages/docker/api/client.py", line 221, in _retrieve_server_version
boefje_1               |     raise DockerException(
boefje_1               | docker.errors.DockerException: Error while fetching server API version: request() got an unexpected keyword argument 'chunked'
boefje_1               |
boefje_1               | The above exception was the direct cause of the following exception:
```

**Comment by ammar92** [comment]
Do you have more logs? `bytes` and `scheduler` are important services that enable boefjes. Also it seems to me that the scheduler is down, hence the connection error?

**Comment by tobiasBDO** [comment]
> Do you have more logs? `bytes` and `scheduler` are important services that enable boefjes. Also it seems to me that the scheduler is down, hence the connection error?

I created a test setup based on main, here attached are my logs directed into a file while setting up openkat and requesting a Subfinder scan for my domain. I can run additional tests in this setup if requested.

[logs.txt](https://github.com/minvws/nl-kat-coordination/files/11487799/logs.txt)

**Comment by ammar92** [comment]
Should be solved in #985 now, I wasn't able to reproduce the error after updating `docker-py`.
